### DISCUSSION TITLE
Preparing data (esp. QSL_SENT) to migrate log to Cloudlog

### DISCUSSION THREAD

**Discussion by raufaser** [magicbug/Cloudlog#2743] [Q&A] [answered]
Hi All,

I just setup Cloudlog and now want to move my logbook from HamOffice to Cloudlog. So i export my entire logbook to ADIF and import it in Cloudlog. 

What are the recommended values QSL_RCVD and QSL_SENT ? At the moment I am using
QSL_SENT: 
- J = Yes 
- N = No (aka the station doesn't do paper cards)
- S = has to be sent

In the logbook I see, that the arrow for "N" is red. But my "N" is more like ignore maybe? And what does invalid stand for? I would just delete invalid QSOs from my log.
And my "S" is more like "queued"?! So should I replace my "S" with "Q"

QSL_RCVD:
- J = Yes
- N = No

All fine here so far, but are there any other values to maybe consider?

Is there a reference which letters Cloudlog uses and what they stand for?

vy 73

**Comment by HB9HIL** [comment]
Hello @raufaser 
The values for QSL_RCVD and QSL_SENT, ... are usually managed by Cloudlog itself. 

I would import the Adif raw as you exported it from Hamoffice. The you can use the Logbook Advanced to edit the QSL Values. You cannot change the Values freely since Cloudlog uses specific values to show correctly on the Browser. Aswell there are more columns in the Database which define how the QSL was sent or received.

<img width="1666" alt="lba" src="https://github.com/magicbug/Cloudlog/assets/80885850/3888b764-9be4-4c44-9f25-ebec0c2b2b6f">

**Comment by HB9HIL** [comment, marked as answer]
If you still want to edit the Values before you import it to Cloudlog (or after updated manually with mysql)

for each QSO in the table "TABLE_HRD_CONTACTS_V01" there are following QSL Related Columns:


**COL_QSL_SENT**
"N" = No  (not sent yet)
"Y" = Yes  (qsl was sent)
"R" = Requested  (the other ham requested a qsl)
"Q" = Queued  (qsl is queued to sent)
"I" = Ignore (Invalid)  (no qsl needed, the ham do not do qsl)

**COL_QSL_SENT_VIA**
"" = empty (can be empty)
"D" = Direct (sent direct to the ham)
"B" = Bureau (sent via Bureau)
"E" = Electronic (sent by E-Mail)
"M" = Sent to QSL Manager

**COL_QSL_RCVD**
"N" = No (no qsl received yet)
"Y" = Yes (qsl received - only Paper QSL)
"R" = Requested (you requested the QSL at the other ham)
"I" = Invalid/Ignore (you wont get an qsl for that)
"V" = Verified (Match) (QSL was verified and the qso is verified)

**COL_QSL_RCVD_VIA**
"" = empty (can be empty)
"D" = Direct (sent direct to the ham)
"B" = Bureau (sent via Bureau)
"E" = Electronic (sent by E-Mail)
"M" = Sent to QSL Manager

Then there are also columns for eQSL, LoTW and so on. The above described QSL Values are only for manually sent/recv QSL's (likely QSL Cards)

Do you need more Info?

**Reply by raufaser** [reply]
That was very helpful.

I just executed some simple SQL statements to update the rows according to your info

For example:
```
UPDATE TABLE_HRD_CONTACTS_V01
SET COL_QSL_SENT = "Q"
WHERE COL_QSL_SENT = "S"
```

Thank you very much for the help.

vy 73

**Reply by HB9HIL** [reply]
You're very welcome 🙂
